ovs/extensions/healthcheck/expose_to_cli.py

```diff
@@ -0,0 +1,324 @@
+        if discovered_data is None:
+            print "Found no method {0} for module {1}".format(method_name, module_name)
+            return cls.print_help(module_name)
+        interrupted = False
+        try:
+            for exposed_module, exposed_methods in discovered_data.iteritems():
+                if module_name and exposed_module != module_name:
+                    continue
+                for method_data in exposed_methods:
+                    if not method_name or method_data['method_name'] == method_name:
+                        mod = imp.load_source(method_data['module_name'], method_data['location'])
+                        cl = getattr(mod, method_data['class'])()
+                        if '--help' in optional_arguments:
+                            print getattr(cl, method_data['function']).__doc__
+                            return
+                        # Execute method
+                        try:
+                            getattr(cl, method_data['function'])(*args)
+                        except Exception:
+                            cls.logger.exception('Error during execution of {0}.{1}'.format(cl, method_data['function']))
+        except KeyboardInterrupt:
+            interrupted = True
+            cls.logger.warning('Caught keyboard interrupt. Output may be incomplete!')
+        if interrupted is False:
+            print "Found no methods for module {0}".format(module_name)
+            return cls.print_help()
+
+    @classmethod
+    def _discover_methods(cls, module_name='', method_name=''):
+        """
+        Discovers all methods with the expose_to_cli decorator
+        :param module_name:  module name specified with the cli
+        :type module_name: str
+        :param method_name: method name specified with the cli
+        :type method_name: str
+        :return: dict that contains the required info based on module_name and method_name
+        :rtype: dict
+        """
+        temp_file_path = '/tmp/_discover_methods'
+        time_format = "%Y-%m-%d %H:%M:%S"
+        version_id = 1
+
+        start_path = cls.START_PATH
+
+        def search_dict(data):
+            """
+            Searches the given data
+            :param data: dict with modules and methods
+            :type data: dict
+            :return: dict with filtered modules and methods
+            :rtype: dict
+            """
+            # Return without expire
+            del data['expires']
+            # Search the dict for the search terms
+            if module_name or method_name:
+                try:
+                    for option in data[module_name]:
+                        if method_name:
+                            if option['method_name'] == method_name:
+                                return {module_name: [option]}
+                        else:
+                            return {module_name: data[module_name]}
+                except KeyError:
+                    pass
```

Comment:
A tiny bit more readable to put the `return None` in between the `pass` and the `else`